Scripts/package.py

```diff
@@ -106,9 +106,18 @@ def download(self):
             print("cached!")
         else:
             flush("downloading {:.2f}MB...".format(self.MB), end=' ')
-            with request.urlopen(self.url) as reader:
-                with open(self.z3_zip, mode="wb") as writer:
-                    writer.write(reader.read())
+            maxTries = 5
+            while maxTries > 0:
+                try:
+                    with request.urlopen(self.url) as reader:
+                        with open(self.z3_zip, mode="wb") as writer:
+                            writer.write(reader.read())
+                    break
+                except (http.client.IncompleteRead) as e:
+                    maxTries -= 1
+                    if maxTries == 0:
+                        raise
```

Comment:
Does this reraise the exception? There should be a way to figure out what went wrong from the log.

Reply:
I see that it does. That's convenient.

Reply:
Yes this seems idiomatic I found, it raises the last thrown exception.